### ISSUE TITLE
Adding links to types that have a global view puts schema into wedged state

### ISSUE THREAD

**Issue by lu-zen** [edgedb/edgedb#4186]
- EdgeDB Version: 
EdgeDB CLI 2.0.1+93a7fab
EdgeDB 2.0+710b779
- OS Version: Linux LTS

Can't create migration on provided schema for:
- removing .email index on member's object (member.esdl);
- removing .email exclusive constraint on member's object;

Steps to Reproduce:

1. Create first migration and apply;
2. remove .email index or exclusive constraint from member's object SDL;
3. create migration;

Schema:
[schema](https://github.com/lu-zen/edgedb-migration-creation-bug/)

**Comment by msullivan** [comment]
The problem is actually much worse than reported: this migration gets into a state where almost everything is broken. (Deleting `current_user` and everything that uses it should work.)

A minimal reproducing schema is:
```
module default {
  global current_user_id -> uuid;
  global current_user := (
    select Member filter .id = global current_user_id
  );

  type Foo {
    link owner := .<avatar[is Member];
  };
  type Member {
    link avatar -> Foo {
      constraint exclusive;
      on source delete delete target if orphan;
      on target delete allow;
    }
  }
}
```

It seems that if we create the global before we create the link, we create a constraint on the link in the derived type for the computed global, but not if we do it in the other order, and this causes a mismatch. Ahhhhh.

**Comment by msullivan** [comment]
@lu-zen You should be able to recover from this state by deleting `current_user` (and whatever references it) and recreating it.

**Comment by lu-zen** [comment]
Hey, thanks!
Tried to remove the references then the globals and the constraint you mentioned remains tho.
```
SchemaError: cannot drop object type 'default::current_user' because other objects in the schema depend on it
  Detail: constraint 'std::exclusive' of link 'avatar' of object type 'default::User' depends on default::current_user[
```

**Comment by msullivan** [comment]
Sigh. Try dropping the constraint too, then recreating the constraint, then the rest of the stuff. Sorry about this one---it's a mess.

I'm working on a fix, and we'll probably try to hurry out a 2.1 that includes it, but that doesn't really save you from needing to recover.

**Comment by lu-zen** [comment]
It's ok, I managed to recover by removing the constraint, tks again!
